### PR TITLE
Adorn appropriate Version.Details.xml dependencies as SourceBuild

### PR DESCRIPTION
**Pull request by MichaelSimons**
Related to https://github.com/dotnet/source-build/issues/2245

runtime, vstest, and nuget.client are wip and do not have source-build intermediate packages available therefore cannot be marked as SourceBuild yet.

### CODE REVIEW

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -69,9 +70,15 @@
       <Uri>https://github.com/dotnet/fsharp</Uri>
       <Sha>13a21d4b08baa0e6b831cc2a7af872aeef8c22a5</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.SourceBuild.Intermediate.fsharp" Version="5.0.2-beta.21321.4">
```
Comment (MichaelSimons) [inline eng/Version.Details.xml:73]: The `Microsoft.FSharp.Compiler` cannot be adorned as source-build as it's version does not match the SB intemerdiate package version therefore the SB Intermediate dependency must be explicitly added.  Same applies to the roslyn-analyzers intermediate dependency that was added.

**Review by sfoslund — APPROVED** [review]

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -236,9 +250,19 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>36b148348ee8312f6369c0c56b0d0fe07deec603</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.SourceBuild.Intermediate.source-build" Version="0.1.0-alpha.1.21318.1">
```
Comment (crummel) [inline eng/Version.Details.xml:253]: Do we need to add a source-build -> SDK subscription for this?

**Review by crummel — APPROVED** [review]

File: eng/Version.Details.xml [hunk 3]
```diff
@@ -104,6 +111,7 @@
     <Dependency Name="Microsoft.NET.ILLink.Tasks" Version="6.0.100-preview.6.21321.1">
       <Uri>https://github.com/mono/linker</Uri>
       <Sha>f2588193553431636b9853b0f87209fa395a72c5</Sha>
+      <SourceBuild RepoName="roslyn" ManagedOnly="true" />
```
Comment (dseefeld) [inline eng/Version.Details.xml:114]: RepoName is incorrect here.

Comment (MichaelSimons) [reply eng/Version.Details.xml]: Good catch - fixed.

File: eng/Version.Details.xml (hunk 2 above)
Comment (MichaelSimons) [reply eng/Version.Details.xml:253]: Yes, I will add post merging.

**Pull request merged**